### DISCUSSION TITLE
failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/bin/sh": stat /bin/sh: no such file or directory: unknown

### DISCUSSION THREAD

**Discussion by D3vil0p3r** [community/community#61493] [Actions] [answered]
### Select Topic Area

Bug

### Body

I have my repo where I host a base container where I would like to automate the building and pushing of it by GitHub Actions.

The repository is this one: https://github.com/Athena-OS/athena-base-docker

The GitHub Actions workflow is: https://github.com/Athena-OS/athena-base-docker/blob/main/.github/workflows/docker-publish.yml

The building and pushing commands work well with no errors when I run them manually on my local machine. When I run them by GitHub Actions, I get error:
```
Step 1/8 : FROM scratch
 ---> 
Step 2/8 : ADD base.tar.zst /
 ---> 36bde3d08ae7
Step 3/8 : RUN ldconfig &&     sed -i '/BUILD_ID/a VERSION_ID=dev' /etc/os-release
 ---> Running in 02aecc849801
failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/bin/sh": stat /bin/sh: no such file or directory: unknown
make: *** [Makefile:78: image-base] Error 1
Error: Process completed with exit code 2.
```
I don't understand why it fails on `ldconfig &&     sed -i '/BUILD_ID/a VERSION_ID=dev' /etc/os-release`, and the error about shim task is weird because if I run this base container manually by `sudo docker run -ti athenaos/base:latest`, `/bin/sh` exists and the command `stat /bin/sh` works with no errors. So I don't understand why I'm getting that error when I use GitHub Actions.

UPDATE: I tried to remove the line related to `ldconfig` and other lines that triggered that strange error, and, since my Workflow use athenaos/base-devel as container, that is the same Im trying to build inside the workflow itself, when I run the workflow the first time (so it uses the base-devel container I built manually locally), it works, so it pushes the updated containers to docker hub. When I run the 2nd time the workflow (so it uses the base-devel container built by GH Actions workflow), it produces this error at the beginning in the **Initialize containers** step:
 ```
 Checking docker version
  /usr/bin/docker version --format '{{.Server.APIVersion}}'
  '1.41'
  Docker daemon API version: '1.41'
  /usr/bin/docker version --format '{{.Client.APIVersion}}'
  '1.41'
  Docker client API version: '1.41'
Clean up resources from previous jobs
  /usr/bin/docker ps --all --quiet --no-trunc --filter "label=c9a4a5"
  /usr/bin/docker network prune --force --filter "label=c9a4a5"
Create local container network
  /usr/bin/docker network create --label c9a4a5 github_network_ddf0c5bf3ff14bbb8b9e[22](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:22)d97faf3ab4
  1aabbff4c9397408b1c3050daca7bd13b2c530a71fb4dbb4853dadeca3776d6a
Starting job container
  /usr/bin/docker pull ***/base-devel:latest
  latest: Pulling from ***/base-devel
  cb531802fa76: Pulling fs layer
  cb531802fa76: Verifying Checksum
  cb531802fa76: Download complete
  cb531802fa76: Pull complete
  Digest: sha256:306371c9d9f3f75bafdb2c37828cffc752[24](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:24)728618b4547ed[25](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:25)3557290cd421d
  Status: Downloaded newer image for ***/base-devel:latest
  docker.io/***/base-devel:latest
  /usr/bin/docker create --name 973038fa6[28](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:29)14d198c5e51c4b501a502_***basedevellatest_d1632d --label c9a4a5 --workdir /__w/athena-base-docker/athena-base-docker --network github_network_ddf0c5bf3ff14bbb8b9e22d97faf3ab4  -e "HOME=/github/home" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work":"/__w" -v "/home/runner/runners/2.[30](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:31)6.0/externals":"/__e":ro -v "/home/runner/work/_temp":"/__w/_temp" -v "/home/runner/work/_actions":"/__w/_actions" -v "/opt/hostedtoolcache":"/__t" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" --entrypoint "tail" ***/base-devel:latest "-f" "/dev/null"
  a95cb5642ceb9ce44a534bec452bf5[11](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:13)a9643976cf[20](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:23)8e8dadd646d65ce08[28](https://github.com/Athena-OS/athena-base-docker/actions/runs/5630998280/job/15257524964#step:2:31)b
  /usr/bin/docker start a95cb5642ceb9ce44a534bec452bf511a9643976cf208e8dadd646d65ce0828b
  Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "tail": executable file not found in $PATH: unknown
  Error: failed to start containers: a95cb5642ceb9ce44a534bec452bf511a9643976cf208e8dadd646d65ce0828b
  Error: Docker start fail with exit code 1
 ```
but `tail` exists in my environment. Indeed, on the first launch (for calling the manually deployed athenaos/base-devel contaiiner) if in the workflow I add
```
      - name: Debug - Check PATH and tail
        run: |
          echo "Current PATH: $PATH"
          which tail
```
I get:
```
Run echo "Current PATH: $PATH"
  
Current PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
/usr/sbin/tail
```
As further evidence, the container base-devel I push to docker hub building it manually, when I pull on my local system it works. The same base-devel container but built on GitHub Actions workflow, when pull it on my local machine (of couse by prune previous image), it returns me the same error:
```
sudo docker run -ti athenaos/base-devel:latest
Unable to find image 'athenaos/base-devel:latest' locally
latest: Pulling from athenaos/base-devel
c521c99e7bb9: Pull complete 
Digest: sha256:6130e4f1f7268f370534c9e53c7459ec958d4d984473de4e4a63b13b14daf8dd
Status: Downloaded newer image for athenaos/base-devel:latest
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/usr/bin/bash": stat /usr/bin/bash: no such file or directory: unknown.
```
So it means that GitHub Actions corrupts the docker image and pushes it to docker hub.

 Why am I getting the error about tail? Is GitHub Actions bugged?

**Comment by D3vil0p3r** [comment, marked as answer]
According to my tests, the cause was the usage of `.tar.zst` format for compressing the rootfs of the container. Indeed, by using:
```
zstd --long -T0 -8 $(1).tar
```
to compress the container in the GitHub Actions workflow, it is not pushed (or built) correctly. Indeed the pushed container is corrupted.

If I replace the command above by compressing the container by xz:
```
xz -9 -T0 -f $(1).tar
```
the container pushed to Docker Hub by the GItHub Actions workflow seem to be working. Indeed by pulling it, it works.

The code of the compression is in [Makefile](https://github.com/Athena-OS/athena-base-docker/blob/main/Makefile).

So, probably, the GitHub environment is not compatible with the usage of `.tar.zst` according to these tests.

**Comment by ZentarXD** [comment]
Pozdrawiam

**Comment by Miskeyceh94** [comment]
Really news code conduct , spesialy answer and mixbettle run rules / rule .. ( rule / rules ) is a "penguasa" . Win / webiner ( web benar ) winer / pemenang indenity spesialy inisial >> "an ( angkit ) , key ( kekey ) , ms , mis (miskey) , Ism ( Isma ) , is , IM saya adalah bos 🖕🤪saya adalah master bos ( boster) energy formula and God , Ch ( CC , ceceh) .

**Comment by RazvanGherlea** [comment]
I am running into this issue as well. Using ECS runner with DING and tied with autoscaling EC2
Please, what flag should I use to overwrite the default defective docker run ?
Thanks in advance
